pom.xml

```diff
@@ -1,21 +1,21 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>org.javassist</groupId>
+  <groupId>ai.h2o</groupId>
   <artifactId>javassist</artifactId>
   <packaging>bundle</packaging>
   <description>
   	Javassist (JAVA programming ASSISTant) makes Java bytecode manipulation
     simple.  It is a class library for editing bytecodes in Java.
   </description>
-  <version>3.27.0-GA</version>
+  <version>3.27.1-h2o-SNAPSHOT</version>
```

Comment:
it might be useful to keep the forked version untouched, and add our internal version, like `3.27.0-h2o-1-SNAPSHOT`, so that it's apparent which is the forked, and have versions of our improvements stated, too

Reply:
@pkozelka good point, I think in this case we don't have to worry because they never update the minor version - it is always 0, there will never official release 3.27.1, the next release will be 3.28.0

wdyt?

Reply:
Yes it definitely makes it non-ambiguous then.
Still, if we maintain patches of multiple external libraries (which I assume is the case), it might be also beneficial to follow the same pattern in custom patch versioning for all of them - as other providers may use different rules for their 3rd place.
Not a huge decision anyway, just trying to think outside of this individual case.

Reply:
Is there a "standard" way of doing this? I found Cloudera uses eg. "3.0.0-cdh6.2.0" - they include their full version.

Reply:
For instance, linux distros use is essentially what I suggested - copy upstream version and add your proprietary version (of the modifications); the exact syntax of the separator/suffix varies a lot. I don't think there is a standard as such; I guess that most improvements are expected to arrive upstream sooner or later (if you submit a PR for it). And for integration tweaks, the ability to upgrade to newer upstream version matters most, which makes visibility of the original version a useful thing.